test/int/framework/frameworkCommonTests.ts

```diff
@@ -125,6 +126,30 @@ export class FrameworkTestSuite {
             await debugClient.continueRequest();
         });
     }
+
+    /**
+     * A generic breakpoint test. This can be used for many different types of breakpoint tests with the following structure:
+     *
+     * 1. Wait for the page to load by waiting for the selector: `waitSelectorId`
+     * 2. Set a breakpoint at `bpLabel`
+     * 3. Execute a trigger event that should cause the breakpoint to be hit using the function `trigger`
+     * 4. Assert that the breakpoint is hit on the expected location, and continue
+     *
+     * @param waitSelector an html selector to identify a resource to wait for for page load
+     * @param bpLabel
+     * @param trigger
+     */
+    genericBreakpointTest(description: string, waitSelector: string, bpLabel: string, trigger: (page: puppeteer.Page) => Promise<void>) {
+        return puppeteerTest(`${this.frameworkName} - ${description}`, this.suiteContext, async (context, page) => {
+            const location = context.breakpointLabels.get(bpLabel);
+            await page.waitForSelector(`${waitSelector}`);
+            await setBreakpoint(this.suiteContext.debugClient, location);
+            const triggerPromise = trigger(page);
+            await this.suiteContext.debugClient.assertStoppedLocation('breakpoint',  location);
```

Comment:
Is it possible to have a race condition here if the stopped event gets sent while we are on line 147?